src/index.ts

```diff
@@ -26,36 +26,43 @@ export function makeMemlet(
 
   // Private methods
 
-  const updateStoreFile = (key: string, data: any, size: number) => {
-    const lastTouchedTimestamp = Date.now()
-
-    const existingFile = store.files[key]
-
-    const newFile: File = {
-      size,
-      filename: key,
+  const addStoreFile = (filename: string, data: any, size: number) => {
+    const file: File = {
+      filename,
       data,
-      lastTouchedTimestamp
+      size,
+      lastTouchedTimestamp: Date.now()
     }
 
-    if (existingFile) {
-      // Update file's position in the file queue
-      fileQueue.requeue(newFile)
-    } else {
-      // Add file to the file queue
-      fileQueue.queue(newFile)
-    }
+    // Add file to file queue
+    fileQueue.queue(file)
+
+    // Add file's size to memory usage
+    adjustMemoryUsage(file.size)
+
+    // Add file to the store files map
+    store.files[filename] = file
+  }
+
+  const updateStoreFile = (filename: string, data: any, size: number) => {
```

Comment:
Remove `updateStoreFile` in favor of inline the two algorithms this method concerns itself with in the two places it is used.